x-pack/plugins/observability/server/services/slo/reset_slo.ts

```diff
@@ -0,0 +1,128 @@
+  getSLOSummaryTransformId,
+  getSLOTransformId,
+  SLO_DESTINATION_INDEX_PATTERN,
+  SLO_MODEL_VERSION,
+  SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+  SLO_SUMMARY_TEMP_INDEX_NAME,
+} from '../../../common/slo/constants';
+import { getSLOSummaryPipelineTemplate } from '../../assets/ingest_templates/slo_summary_pipeline_template';
+import { retryTransientEsErrors } from '../../utils/retry';
+import { SLORepository } from './slo_repository';
+import { createTempSummaryDocument } from './summary_transform_generator/helpers/create_temp_summary';
+import { TransformManager } from './transform_manager';
+
+export class ResetSLO {
+  constructor(
+    private esClient: ElasticsearchClient,
+    private repository: SLORepository,
+    private transformManager: TransformManager,
+    private summaryTransformManager: TransformManager,
+    private logger: Logger,
+    private spaceId: string
+  ) {}
+
+  public async execute(sloId: string) {
+    const slo = await this.repository.findById(sloId);
+
+    const summaryTransformId = getSLOSummaryTransformId(slo.id, slo.revision);
+    await this.summaryTransformManager.stop(summaryTransformId);
+    await this.summaryTransformManager.uninstall(summaryTransformId);
+
+    const rollupTransformId = getSLOTransformId(slo.id, slo.revision);
+    await this.transformManager.stop(rollupTransformId);
+    await this.transformManager.uninstall(rollupTransformId);
+
+    await Promise.all([this.deleteRollupData(slo.id), this.deleteSummaryData(slo.id)]);
+
+    try {
+      await this.transformManager.install(slo);
+      await this.transformManager.start(rollupTransformId);
+      await retryTransientEsErrors(
+        () => this.esClient.ingest.putPipeline(getSLOSummaryPipelineTemplate(slo, this.spaceId)),
+        { logger: this.logger }
+      );
+
+      await this.summaryTransformManager.install(slo);
+      await this.summaryTransformManager.start(summaryTransformId);
+
+      await retryTransientEsErrors(
+        () =>
+          this.esClient.index({
+            index: SLO_SUMMARY_TEMP_INDEX_NAME,
+            id: `slo-${slo.id}`,
+            document: createTempSummaryDocument(slo, this.spaceId),
+            refresh: true,
+          }),
+        { logger: this.logger }
+      );
+    } catch (err) {
+      this.logger.error(
+        `Cannot reset the SLO [id: ${slo.id}, revision: ${slo.revision}]. Rolling back.`
+      );
+
+      await this.summaryTransformManager.stop(summaryTransformId);
+      await this.summaryTransformManager.uninstall(summaryTransformId);
+      await this.transformManager.stop(rollupTransformId);
+      await this.transformManager.uninstall(rollupTransformId);
+      await this.esClient.ingest.deletePipeline(
+        { id: getSLOSummaryPipelineId(slo.id, slo.revision) },
+        { ignore: [404] }
+      );
+
+      throw err;
+    }
+
+    const updatedSlo = await this.repository.save(
+      Object.assign({ ...slo, version: SLO_MODEL_VERSION, updatedAt: new Date() })
```

Comment:
There's only one parameter passed to `Object.assign` here.

Did you mean 
```suggestion
      Object.assign(slo, { version: SLO_MODEL_VERSION, updatedAt: new Date() })
```

Reply:
Actually the Object.assign is useless. What I want to achieve here is creating a copy of the original slo, while updating the version and updatedAt. But I do not want to change the original slo. Doing `Object.assign(slo, { ... })` would mutate the original slo